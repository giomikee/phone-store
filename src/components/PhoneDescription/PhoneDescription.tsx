import { Box, List, ListItem, ListItemText, Skeleton, Stack, Typography } from '@mui/material';
import type { Phone, Specs } from '../../interfaces';
import { HEADINGS, SPECS_TRANSLATIONS } from '../../constants';

const PhoneDescription = ({phoneDetails}: {phoneDetails: Phone | null}) => {
    if (!phoneDetails) {
        return <Skeleton height={500} />
    }
    
    return (
        <Stack spacing={2}>
            <Box>
                <Typography 
                component={HEADINGS.h3}
                variant={HEADINGS.h6}
                color='info'
                >
                Descripci√≥n
                </Typography>
                
                <Typography component='p' variant='body1'>
                {phoneDetails.description}
                </Typography>
            </Box>
            
            <Box>
                <Typography 
                component={HEADINGS.h3}
                variant={HEADINGS.h6}
                color='info'
                >
                Especificaciones
                </Typography>
                
                <List dense>
                {Object.keys(phoneDetails.specs!).map((specKey) => (
                    <ListItem key={specKey}>
                    <ListItemText
                    primary={SPECS_TRANSLATIONS[specKey as keyof typeof SPECS_TRANSLATIONS]}
                    secondary={phoneDetails.specs![specKey as keyof Specs]}
                    />
                    </ListItem>
                ))}
                </List>
            </Box>
        </Stack>
    )
}

export default PhoneDescription