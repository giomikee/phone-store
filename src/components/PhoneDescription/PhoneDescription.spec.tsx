import { describe, expect, it } from 'vitest';
import {  render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import PhoneDescription from './PhoneDescription';

describe('Given PhoneDescription component', () => {
    const defaultProps = {
        phoneDetails: {
            id: '1',
            brand: 'Apple',
            name: 'iPhone 15',
            description: 'description',
            basePrice: 999,
            imageUrl: 'imageUrl',
            specs: {
                screen: 'screen',
                resolution: 'resolution',
                processor: 'processor',
                mainCamera: 'mainCamera',
                selfieCamera: 'selfieCamera',
                battery: 'battery',
                os: 'os',
                screenRefreshRate: 'screenRefreshRate',
            },
        },
    };

    describe.each([
        defaultProps.phoneDetails,
        null,
    ])('When the component is rendered', (phoneDetails) => {
        describe(`And phoneDetails is ${phoneDetails ? 'not null' : 'null' }`, () => {

            it('Then it should match the snapshot', () => {
                const { container } = render(<PhoneDescription {...defaultProps} phoneDetails={phoneDetails} />);
                expect(container).toMatchSnapshot();
            });
        })
    });
});