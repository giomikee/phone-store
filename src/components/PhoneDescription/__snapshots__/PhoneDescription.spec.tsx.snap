// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given PhoneDescription component > When the component is rendered > And phoneDetails is not null > Then it should match the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1sazv7p-MuiStack-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h3
        class="MuiTypography-root MuiTypography-h6 css-1hgt7rk-MuiTypography-root"
      >
        Descripci칩n
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        description
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h3
        class="MuiTypography-root MuiTypography-h6 css-1hgt7rk-MuiTypography-root"
      >
        Especificaciones
      </h3>
      <ul
        class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Pantalla
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              screen
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Resoluci칩n
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              resolution
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Procesador
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              processor
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              C치mara trasera
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              mainCamera
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              C치mara delantera
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              selfieCamera
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Bateria
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              battery
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Sistema Operativo
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              os
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-qyw6ft-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
            >
              Refresh rate
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
            >
              screenRefreshRate
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Given PhoneDescription component > When the component is rendered > And phoneDetails is null > Then it should match the snapshot 1`] = `
<div>
  <span
    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
    style="height: 500px;"
  />
</div>
`;
